psi4/driver/procrouting/proc.py

```diff
@@ -988,6 +989,46 @@ def select_mp4(name, **kwargs):
         return func(name, **kwargs)
 
 
+def select_adc2(name, **kwargs):
+    """Function selecting the algorithm for ADC(2) excited state energy
+    call and directing to specified or best-performance default modules.
+
+    """
+    reference = core.get_option('SCF', 'REFERENCE')
+    mtd_type = core.get_global_option('MP_TYPE')  # TODO or ADC_TYPE or CC_TYPE or ...
```

Comment:
different names are mostly for enough knobs to influence single jobs with complex formulation like cbs. mp3 and adc seem pretty orthogonal, so `MP_TYPE` sounds good for now.